src/Chess/ChessBoard.hs

```diff
@@ -31,7 +31,28 @@ chessBoardRowAsString rowNum chessBoard = rowNumStr ++ "  " ++ (foldl (\ rowStr
 -- WARNING: Assumes given move is valid!
 -- TODO: implement + test this (1.5 hours) [Yiyi]
 makeMove :: ChessBoard -> ChessMove -> ChessBoard
-makeMove chessBoard move = chessBoard
+makeMove cb@(ChessBoard pieces state) (ChessMove from to) = 
+    ChessBoard (putPiece pieceToMove to (removePiece from pieces)) (if (isCheckMate cb (getPieceColour pieceToMove)) then Over else state)
+    where pieceToMove = fromJust(getPieceAt from cb)
+
+
+-- Removes the piece at the given position from the ChessBoard if it exists
+-- Does nothing if the position is empty
+removePiece :: ChessPosition -> [(ChessPosition, ChessPiece)] -> [(ChessPosition, ChessPiece)]
+removePiece chessPosition board = 
+    if (lookup chessPosition board == Nothing)
```

Comment:
the `if-else` statement isn't required. If there is no piece at the position, then `filter` will just return the same pieces as before.